src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java

```diff
@@ -418,4 +475,30 @@ private NumberRange getOperationRange(ImageWorker iw) {
             }
         }
     }
+
+    private String getPercentagesLabelPortion(double[] percentages, int i) {
+        if (percentages == null) return "";
+        else return " (" + percentages[i] + "%)";
+    }
+
+    private double[] getPercentagesForCustom(RenderedImage image, int intervals) {
```

Comment:
This seems to assume an equal interval approach, but with the custom classification the client is passing in the break values instead, they can be distributed in whatever way. User provided breaks should be used, please add a test with custom values and a non uniformly distributed set of breaks.

Reply:
fixed